src/protocol.h

```diff
@@ -55,220 +55,255 @@ class CMessageHeader
 
 /**
  * Bitcoin protocol message types. When adding new message types, don't forget
- * to update allNetMessageTypes in protocol.cpp.
+ * to update ALL_NET_MESSAGE_TYPES below.
  */
 namespace NetMsgType {
-
 /**
  * The version message provides information about the transmitting node to the
  * receiving node at the beginning of a connection.
  */
-extern const char* VERSION;
+inline constexpr const char* VERSION{"version"};
```

Comment:
Might as well make these `std::string`? Or is there still a good reason that these are `char*`? (it used to be that the message-making functions took char*, but i don't think that's the case anymore)

Reply:
IIRC this was left because `constexpr std::string` was not available, but perhaps it is now in c++20?

Reply:
i mean it's defining an array of `std::string` constexpr below it, so i would expect single std::strings to be possible too, i haven't tried it though

Reply:
This was tried: https://github.com/bitcoin/bitcoin/pull/29421#discussion_r1491140321

Here is a draft that changes to `std::string_view` the following:

* `NetMsgType::*`
* `ALL_NET_MESSAGE_TYPES`
* `CMessageHeader:pchCommand` (renamed to `CMessageHeader:m_command`)
* `CNetMessage::m_type`
* `V2_MESSAGE_IDS`
* `V2MessageMap::m_map`
* `V2Transport::m_send_type`
and the functions that handle message types.

The scope of this can be reduced by converting the `std::string_view` to `std::string` at some point. The previous code uses `std::string` (or `char*`) everywhere.

https://github.com/vasild/bitcoin/commits/NetMsgType_string_view/